normation/rudder/roles/manage_rudder_repository/molecule/with_credentials/debian.yml

```diff
@@ -0,0 +1,16 @@
+---
+- name: Apt repo key must be imported
+  ansible.builtin.apt_key:
+    state: present
+    url: "{{ apt_key_url }}"
+  register: repo_key
+  failed_when: (repo_key is changed) or (repo_key is failed)
+
+- name: Repository must be correct
+  ansible.builtin.apt_repository:
+    state: present
+    repo: "deb https://my_user:my_password@download.rudder.io/apt/6.2 {{ ansible_distribution_release }} main"
```

Comment:
It might be better to call the values either via vault or via templating jinja.

Reply:
Like : deb https://{rudder-username}:{rudder-password}@download.rudder.io/apt/6.2